lib/Dialect/FIRRTL/Transforms/IMConstProp.cpp

```diff
@@ -603,6 +596,12 @@ void IMConstPropPass::visitStrictConnect(StrictConnectOp connect) {
       return;
   }
 
+  // Make aggregates overdefined for now.  Fix when context sensitive.
+  if (isAggregate(dest.getOwner())) {
```

Comment:
Probably it would be better to check whether `dest` has an aggregate type as well, i.e:
`isAggregate(dest.getOwner()) || !dest.getType().cast<FIRRTLType>().isGrond()`.

Or it might be more verbose to keep to reject bundle level connections for now.

Reply:
The idea is to have any bundle thing reduce to overdefined, but processed.  This is the minimum required for safely handling aggregates.  Then your PR with field sensitivity can fix this pessimism.